app/src/main/java/com/example/iou/bill/adapters/BillItemAdapter.java

```diff
@@ -85,7 +87,12 @@ public void bind(BillItem bill) {
                 checkbox.setOnClickListener(new View.OnClickListener() {
                     @Override
                     public void onClick(View v) {
-                        bill.toggleCheckbox(finalI, checkbox.isChecked());
+
+                        // Set the UI to display the check or uncheck
+                        checkbox.setChecked(!bill.getChecks(finalI));
+
+                        // Update the BillItem bill
+                        bill.toggleCheckbox(finalI, !bill.getChecks(finalI));
```

Comment:
A little bit of a nit, but we can make consolidate some logic here

`boolean checked = !bill.getChecks(finalI);`
 `checkbox.setChecked(checked);`
 `bill.toggleCheckBox(finalI, checked);`